### PR TITLE
Purity

### PR DESCRIPTION
**Pull request by wbars**

### CODE REVIEW

File: bz2/bz2.php [hunk 1]
```diff
@@ -30,6 +33,7 @@ function bzopen ($file, $mode) {}
  * </p>
  * @return string the uncompressed data, or <b>FALSE</b> on error.
  */
+#[Pure]
```
Comment (wbars) [inline bz2/bz2.php:36]: This will change file pointer
```suggestion

```

File: date/date.php [hunk 2]
```diff
@@ -1041,6 +1063,7 @@ function date_timezone_get ($object) {}
  * @return DateTime|false <p>Returns the
  * {@see DateTime} object for method chaining or <b>FALSE</b> on failure.</p>
  */
+#[Pure]
```
Comment (wbars) [inline date/date.php:1066]: Modifying DateTime state?
```suggestion
```

File: date/date.php [hunk 3]
```diff
@@ -1075,6 +1100,7 @@ function date_diff ($baseObject, $targetObject, $absolute = false) {}
...
```
Comment (wbars) [inline date/date.php:1103]: Modifying DateTime state?
```suggestion
```

File: date/date.php [hunk 4]
```diff
@@ -1092,6 +1118,7 @@ function date_time_set ($object, $hour, $minute, $second = 0, $microsecond = 0)
  * {@see DateTime} object for method chaining or <b>FALSE</b> on failure.
  * </p>
  */
+#[Pure]
```
Comment (wbars) [inline date/date.php:1121]: Modifying DateTime state?
```suggestion
```

File: date/date.php [hunk 5]
```diff
@@ -1106,6 +1133,7 @@ function date_date_set ($object, $year, $month, $day) {}
  * Returns the {@see DateTime} object for method chaining or <strong><code>FALSE</code></strong> on failure.
  * </p>
  */
+#[Pure]
```
Comment (wbars) [inline date/date.php:1136]: Modifying DateTime state?
```suggestion
```

File: date/date.php [hunk 6]
```diff
@@ -1118,6 +1146,7 @@ function date_isodate_set ($object, $year, $week, $day = 1) {}
  * @return DateTime|false
  * {@see DateTime} object for call chaining or <b>FALSE</b> on failure
  */
+#[Pure]
```
Comment (wbars) [inline date/date.php:1149]: Modifying DateTime state?
```suggestion
```

File: http/http.php [hunk 7]
```diff
@@ -2184,6 +2188,7 @@ function http_build_str (array $query, $prefix = null, $arg_separator = null) {}
  * </p>
  * @return string the negotiated language or the default language (i.e. first array entry) if none match.
  */
+#[Pure]
```
Comment (wbars) [inline http/http.php:2191]: Pass-by-ref parameter can be modified
```suggestion
```

File: http/http.php [hunk 8]
```diff
@@ -2198,6 +2203,7 @@ function http_negotiate_language (array $supported,  array &$result = null ) {}
  * </p>
  * @return string the negotiated charset or the default charset (i.e. first array entry) if none match.
  */
+#[Pure]
```
Comment (wbars) [inline http/http.php:2206]: Pass-by-ref parameter can be modified
```suggestion
```

File: http/http.php [hunk 9]
```diff
@@ -2212,6 +2218,7 @@ function http_negotiate_charset (array $supported,  array &$result = null ) {}
  * </p>
  * @return string the negotiated content type or the default content type (i.e. first array entry) if none match.
  */
+#[Pure]
```
Comment (wbars) [inline http/http.php:2221]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 10]
```diff
@@ -3536,6 +3550,7 @@ function numfmt_format_currency(NumberFormatter $fmt, $value, $currency) { }
  * </p>
  * @return float|false The parsed numeric value or <b>FALSE</b> on error.
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:3553]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 11]
```diff
@@ -4344,6 +4406,7 @@ function datefmt_format_object($object, $format = null, $locale = null) { }
  * </p>
  * @return int|float|false timestamp parsed value
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:4409]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 12]
```diff
@@ -4362,6 +4425,7 @@ function datefmt_parse(IntlDateFormatter $df, $value, &$position = null) { }
  * </p>
  * @return array|false Localtime compatible array of integers : contains 24 hour clock value in tm_hour field
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:4428]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 13]
```diff
@@ -4572,6 +4646,7 @@ function grapheme_stristr($haystack, $needle, $before_needle = false) { }
  * @return string|false A string starting at offset $start and ending on a default grapheme cluster
  * boundary that conforms to the $size and $extract_type specified.
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:4649]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 14]
```diff
@@ -4600,6 +4675,7 @@ function grapheme_extract($haystack, $size, $extract_type = null, $start = 0, &$
  * </p>
  * @return string|false The ACE encoded version of the domain name or <b>FALSE</b> on failure.
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:4678]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 15]
```diff
@@ -4628,6 +4704,7 @@ function idn_to_ascii($domain, $options = 0, $variant = INTL_IDNA_VARIANT_UTS46,
  * RFC 3490 4.2 states though "ToUnicode never fails. If any step fails, then the original input
  * sequence is returned immediately in that step."
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:4707]: Pass-by-ref parameter can be modified
```suggestion
```

File: intl/intl.php [hunk 16]
```diff
@@ -4758,6 +4841,7 @@ function intlcal_get_time($calendar) { }
  * Returns <b>TRUE</b> on success and <b>FALSE</b> on failure.
  * @since 5.5
  */
+#[Pure]
```
Comment (wbars) [inline intl/intl.php:4844]: Setter?
```suggestion
```

File: mbstring/mbstring.php [hunk 17]
```diff
@@ -990,6 +1024,7 @@ function mb_ereg_replace ($pattern, $replacement, $string, $options = "msr") {}
  * </p>
  * @since 5.4.1
  */
+#[Pure]
```
Comment (wbars) [inline mbstring/mbstring.php:1027]: Callback parameter my have side effects
```suggestion
```

File: pcre/pcre.php [hunk 18]
```diff
@@ -154,6 +155,7 @@
  * matches given <i>subject</i>, 0 if it does not, or <b>FALSE</b>
  * if an error occurred.
  */
+#[Pure]
```
Comment (wbars) [inline pcre/pcre.php:158]: Pass-by-ref parameter may be modified
```suggestion
```

File: pcre/pcre.php [hunk 19]
```diff
@@ -214,6 +216,7 @@ function preg_match ($pattern, $subject, array &$matches = null, $flags = 0, $of
  * @return int|false the number of full pattern matches (which might be zero),
  * or <b>FALSE</b> if an error occurred.
  */
+#[Pure]
```
Comment (wbars) [inline pcre/pcre.php:219]: Pass-by-ref parameter may be modified
```suggestion
```

File: pcre/pcre.php [hunk 20]
```diff
@@ -373,6 +377,7 @@ function preg_replace ($pattern, $replacement, $subject, $limit = -1, &$count =
  * If matches are found, the new subject will be returned, otherwise
  * <i>subject</i> will be returned unchanged.
  */
+#[Pure]
```
Comment (wbars) [inline pcre/pcre.php:380]: Callback parameter may have side effects
```suggestion
```

File: pcre/pcre.php [hunk 21]
```diff
@@ -386,6 +391,7 @@ function preg_replace_callback ($pattern, callable $callback, $subject, $limit =
  * @return string|string[]|null  <p>preg_replace_callback_array() returns an array if the subject parameter is an array, or a string otherwise. On errors the return value is NULL</p>
  * <p>If matches are found, the new subject will be returned, otherwise subject will be returned unchanged.</p>
  */
+#[Pure]
```
Comment (wbars) [inline pcre/pcre.php:394]: Pass-by-ref parameter may be modified
```suggestion
```

File: pcre/pcre.php [hunk 22]
```diff
@@ -404,6 +410,7 @@ function preg_replace_callback_array ($pattern, $subject , $limit = -1, &$count
  * is returned when <i>subject</i> is an array
  * or <b>NULL</b> otherwise.
  */
+#[Pure]
```
Comment (wbars) [inline pcre/pcre.php:413]: Pass-by-ref parameter may be modified
```suggestion
```

File: pcre/pcre.php [hunk 23]
```diff
@@ -302,6 +305,7 @@ function preg_match_all ($pattern, $subject, array &$matches = null, $flags = PR
  * be returned, otherwise <i>subject</i> will be
  * returned unchanged or <b>NULL</b> if an error occurred.
  */
+#[Pure]
```
Comment (wbars) [inline pcre/pcre.php:308]: Pass-by-ref parameter may be modified
```suggestion
```

File: standard/standard_0.php [hunk 24]
```diff
@@ -866,6 +883,7 @@ function iptcembed ($iptc_data, $filename, $spool = null) {}
  * <p>
  * On failure, false is returned.
  */
+#[Pure]
```
Comment (wbars) [inline standard/standard_0.php:886]: Pass-by-ref parameter may be modified
```suggestion
```

File: zlib/zlib.php [hunk 25]
```diff
@@ -74,6 +76,7 @@ function gzgetc ($stream) {}
  * </p>
  * @return string|false The uncompressed string, or <b>FALSE</b> on error.
  */
+#[Pure]
```
Comment (wbars) [inline zlib/zlib.php:79]: Modifies file pointer?
```suggestion
```

File: zlib/zlib.php [hunk 26]
```diff
@@ -60,6 +61,7 @@ function gzeof ($stream) {}
  * </p>
  * @return string|false The uncompressed character or <b>FALSE</b> on EOF (unlike <b>gzeof</b>).
  */
+#[Pure]
```
Comment (wbars) [inline zlib/zlib.php:64]: Modifies file pointer?
```suggestion
```

File: zlib/zlib.php [hunk 27]
```diff
@@ -94,6 +97,7 @@ function gzgets ($stream, $length) {}
  * @deprecated 7.3
  * @removed 8.0
  */
+#[Pure]
```
Comment (wbars) [inline zlib/zlib.php:100]: Modifies file pointer?
```suggestion
```

File: zlib/zlib.php [hunk 28]
```diff
@@ -108,6 +112,7 @@ function gzgetss ($zp, $length, $allowable_tags = null) {}
  * </p>
  * @return string The data that have been read.
  */
+#[Pure]
```
Comment (wbars) [inline zlib/zlib.php:115]: Modifies file pointer?
```suggestion
```

File: zlib/zlib.php [hunk 29]
```diff
@@ -184,6 +189,7 @@ function gzseek ($stream, $offset, $whence = SEEK_SET) {}
  * </p>
  * @return int|false The position of the file pointer or <b>FALSE</b> if an error occurs.
  */
+#[Pure]
```
Comment (wbars) [inline zlib/zlib.php:192]: Modifies file pointer?
```suggestion
```

File: standard/standard_8.php [hunk 30]
```diff
@@ -92,6 +94,7 @@ function header_register_callback ( callable $callback ) {}
  * that can be used directly in an IMG tag.<br>
  * On failure, FALSE is returned.
  */
+#[Pure]
```
Comment (MaXal) [inline standard/standard_8.php:97]:
```suggestion
```

Comment (MaXal) [reply standard/standard_8.php:97]: Pass by ref

**Review by MaXal — APPROVED** [review]

**Pull request merged**